ReusableViews/ReusableBtn.swift

```diff
@@ -0,0 +1,62 @@
+//
+//  ReusableBtn.swift
+//  WeatherDemoApp
+//
+//  Created by ahmed on 03/04/2024.
+//
+
+import UIKit
+
+final class ReusableButton: UIButton {
+    
+    var btnTappedAction: VoidCallback?
```

Comment:
```suggestion
    private var buttonTapped: VoidCallback?
```
1. Please don't use abbreviations in Swift. 
2. buttonTapped - onButtonTap - tapAction are all good names. 
3. Private properties should be kept private.